clvm_runtime/curry.clvm

```diff
@@ -7,15 +7,15 @@
   ;; (curry add2 15) ; this yields a function that accepts ONE argument, and adds 15 to it
 
   ; (curry_args sum (list 50 60)) => returns a function that is like (sum 50 60 ...)
-  (defun curry_args (func list_of_args) (qq ((c (q (unquote func)) (unquote (fix_curry_args list_of_args (q 1)))))))
+  (defun curry_args (func list_of_args) (qq ((c (q . (unquote func)) (unquote (fix_curry_args list_of_args (q . 1)))))))
 
   ;; (curry sum 50 60) => returns a function that is like (sum 50 60 ...)
   (defun curry (func . args) (curry_args func args))
 
   ;; utility function used by `curry_args`
   (defun fix_curry_args (items core)
     (if items
-        (qq (c (q (unquote (f items))) (unquote (fix_curry_args (r items) core))))
+        (qq (c (q . (unquote (f items))) (unquote (fix_curry_args (r items) core))))
```

Comment:
I was under the impression that the quote cons-encoding was a property of how `(q ...)` was *lowered* into the low level language, alternatively how it was serialised. This makes it look like it bubbles up all the way to the high level language. Is that the intention?

Reply:
It would be a crime against humanity to display `(q 2)` when we mean `(q . 2)`